### PR TITLE
DROOLS-4211 / DROOLS-4212: Search feature - Implement search mechanism PoC on GDT / DMN graph

### PR DESCRIPTION
**Pull request by karreiro**
### JIRAs:

- https://issues.jboss.org/browse/DROOLS-4211
- https://issues.jboss.org/browse/DROOLS-4212

---

### Part of an ensemble:
- https://github.com/kiegroup/appformer/pull/738
- https://github.com/kiegroup/kie-wb-common/pull/2749
- https://github.com/kiegroup/drools-wb/pull/1191

---

### Description
DROOLS-4211 and DROOLS-4212 JIRAs comprehend only the search mechanism. Thus, the PR implements that for Guided Decision Tables, DMN graphs, and DMN boxed expressions, without a rich user experience (which will be implemented on [DROOLS-4306](https://issues.jboss.org/browse/DROOLS-4306)).
 
To test this PR, just execute the following code into the _browser inspector_:
 
```javascript
document.querySelector('.kie-search-bar').style.display = ""
```

The search component will be enabled, and now you can type something and press <kbd>Enter</kbd> to trigger the search.

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/search/GuidedDecisionTableSearchableElement.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    public String getValue() {
+        return convertDTCellValueToString(cellValue52);
+    }
+
+    @Override
+    public Command onFound() {
+        return () -> highlightHelper()
+                .withMinX(getMinX())
+                .withMinY(getMinY())
+                .withPaddingX(200)
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/search/GuidedDecisionTableSearchableElement.java]: Can you please explain these arbitrary padding values?

Comment (karreiro) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/editor/search/GuidedDecisionTableSearchableElement.java]: Fixed. Thanks!

**Comment by karreiro** [comment]
Thank you for the review, @manstis. Changes applied :-)

**Comment by karreiro** [comment]
(rebasing the branch... 👀)

**Comment by karreiro** [comment]
Jenkins please retest this.

**Pull request merged**